### PR TITLE
Upgrade to React 16

### LINKED ISSUE #823: Upgrade Reaper UI to React 16.8+
While we rely on React 15 in Reaper, it is blocking us from using modern components and dependencies which now often rely on React >= 16.8.

This change will impact most of the jsx code and the webpack configuration as our code is mostly incompatible with the latest versions of React. Also, the hot loader will need to be upgraded, which will require more changes to both the jsx code and the webpack config.

We're targeting this change for Reaper 3.0.

### PR DESCRIPTION
**Pull request by ossarga**
#823 
* Updated package.json components and dependencies to work with React 16.
* Removed unused file-loader module from package.json.
* Updated JSX component files so that React.createClass replaces create-react-class and React.PropTypes replaces prop-types.
* Replaced ReactTags component and dropdown select box with ReactSelect component.
* Removed unused ReactTags CSS definitions.
* Replaced react-widgets DatTime component with react-datepicker component.
* Updated RepairForm component to gather details for both on demand and scheduled repairs. Removed Scheduled repairs form.
* Updated Webpack to allow Reaper host to be specified as a commandline argument.
* Added script to build navbar.jsx and added script command to package.json.
* Moved SegmentModal into RepairList render function and changed CSS so that the modal is always opaque.
* Fixed props and styling on buttons in SegmentList.
* Changed row data that was using disabled buttons (with no click handler) to be text data with a styling.

### CODE REVIEW

**Comment by adejanovski** [comment]
I fixed modal windows that were not sizing properly by downgrading react-bootstrap to 0.33.1. v1.0.0 relies on Boostrap 4 and we're still using Bootstrap 3.

We had some parts using the formerly used React Tags styling to display lists of nodes and tables.
Since this is gone now, rendering looks bad:

127 0 0 1 8000 webpack-dev-server  (17)
127 0 0 1 8000 webpack-dev-server  (14)

It looks like the progress bars on the repair screen lost their colors, which showed the status of the repair:
127 0 0 1 8000 webpack-dev-server  (15)

File: src/ui/build-navbar.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Copyright 2017-2019 The Last Pickle Ltd
```
Comment (adejanovski) [inline src/ui/build-navbar.sh:2]: I guess we should now start putting Datastax copyrights on the files we touch.

File: src/ui/webpack.config.js [hunk 2]
```diff
@@ -169,13 +190,14 @@ module.exports = {
       }
     }
   },
+  mode: isDev === "1" ? "development" : "production",
```
Comment (adejanovski) [inline src/ui/webpack.config.js]: shouldn't it be `mode: isDev ? "development" : "production",` ?

**Review by adejanovski — APPROVED** [review]
I have a few minor questions and nits, but it LGTM!

Super happy to get that merged in :shipit:

File: src/ui/app/jsx/cluster-list.jsx [hunk 3]
```diff
@@ -43,7 +45,7 @@ const Cluster = React.createClass({
           };
   },
 
-  componentWillMount: function() {
+  UNSAFE_componentWillMount: function() {
```
[the remark this answers is not in the document]
Comment (ossarga) [reply src/ui/app/jsx/cluster-list.jsx:48]: The path to migration is outlined in this blog post: https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html. We need to raise an issue for this.

[the remark this answers is not in the document]
Comment (ossarga) [reply src/ui/app/jsx/cluster-list.jsx:48]: Raised #909 to capture this issue.

File: src/ui/app/jsx/event-screen.jsx [hunk 4]
```diff
@@ -23,46 +25,46 @@ import DiagEventsSubscriptionForm from "jsx/event-subscription-form";
...
 
     this._listenSubscription = this.props.listenSubscriptionSubject.subscribeOnNext(subscription => {
 
       if(this.state.activeEventSource) {
         this.state.activeEventSource.close();
       }
 
-      const URL_PREFIX = getUrlPrefix(window.top.location.pathname);
-      const source = new EventSource(`${URL_PREFIX}/diag_event/sse_listen/${subscription.id}`);
+      const urlPrefix = getUrlPrefix(window.top.location.pathname);
```
[the remark this answers is not in the document]
Comment (ossarga) [reply src/ui/app/jsx/event-screen.jsx:66]: I think for globals yes. It also depends on the style guide we are using. I think this should be fine according to the Google JS style guide on [variable](https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations) declarations.

File: src/ui/app/jsx/repair-form.jsx [hunk 5]
```diff
@@ -66,93 +112,125 @@ const repairForm = React.createClass({
...
 
-  _getSuggestions: function() {
-    var nodes = this.state.clusterStatus.nodes_status.endpointStates[0].endpointNames;
-    nodes.sort();
-    this.state.nodeSuggestions = nodes;
-    
-    var datacenters = Object.keys(this.state.clusterStatus.nodes_status.endpointStates[0].endpoints);
-    datacenters.sort;
-    this.setState({datacenterSuggestions: datacenters});
+  _getDatacetnerOptions: function() {
```
[the remark this answers is not in the document]
Comment (ossarga) [reply src/ui/app/jsx/repair-form.jsx]: Nicely spotted. Fixed.

File: src/ui/app/jsx/schedule-screen.jsx [hunk 6]
```diff
@@ -74,7 +76,7 @@ const ScheduleScreen = React.createClass({
 
       <div className="row">
                 <div className="col-lg-12">
-                  <ScheduleForm addScheduleSubject={this.props.addScheduleSubject} addScheduleResult={this.props.addScheduleResult} clusterNames={this.props.clusterNames} currentCluster={this.props.currentCluster}> </ScheduleForm>
+                  <RepairForm addRepairSubject={this.props.addRepairSubject} addRepairResult={this.props.addRepairResult} clusterNames={this.props.clusterNames} currentCluster={this.props.currentCluster} formType="schedule"> </RepairForm>
```
[the remark this answers is not in the document]
Comment (ossarga) [reply src/ui/app/jsx/schedule-screen.jsx:79]: Thanks 😄

File: src/ui/build-navbar.sh (hunk 1 above)
Comment (ossarga) [reply src/ui/build-navbar.sh:2]: Raised issue #910 to fix this across the entire repo.

File: src/ui/webpack.config.js (hunk 2 above)
Comment (ossarga) [reply src/ui/webpack.config.js]: Yes, well spotted. We also need to the line where `isDev` is declared. It should be:
```
const isDev = process.argv.indexOf('-d') !== -1 || !!+process.env.BUILD_DEV;
```
Will fix this.

File: src/ui/app/jsx/component.jsx [hunk 7]
```diff
@@ -14,8 +14,9 @@
 //  limitations under the License.
 
 import React from "react";
+import CreateReactClass from 'create-react-class';
```
[the remark this answers is not in the document]
Comment (ossarga) [reply src/ui/app/jsx/component.jsx]: Agreed. Removed _src/ui/app/jsx/dashboard-page.jsx_ as well as it appear to be unused.

**Pull request merged**